### ISSUE TITLE
Add a new community-level userrole with web access

### ISSUE THREAD

**Issue by DavidBaldsiefen** [hzi-braunschweig/SORMAS-Project#4290]
### Situation Description

A lot of german users need a community-level user with access to the webapp.

### Feature Description

For now, add a new user role. I will complete this issue with the required userrights asap.

Required userrights:

- listing cases & contacts in their community
- creating and sending quarantine documents & subsequently change the related quarantine order date
- Editing the case-person and contact-person

Creating new Cases and Contacts does not seem to be a requirement.

### Possible Alternatives

Waiting for the new user rights model is not an option, due to this issues urgency.

**Comment by StefanKock** [comment]
Duplicate to #1960

**Comment by DavidBaldsiefen** [comment]
Reopened due to new requirements not covered with #1960

**Comment by DavidBaldsiefen** [comment]
I've discussed this further with the user, and we decided that it would be best (for now) if the community officer has more rights than the community informant. This way we won't have to create a new userrole every time somebody requires a community-level role to have some additional right

**Comment by bernardsilenou** [comment]
Suggestions:
- we can call this role "community surveillance officer" 
- In order not to create another community contact officer role, we can  give this role the "community contact officer" rights.
-  they can have the combination of surveillance officer and contact officer rights but limited to the community of the user
- Since the community is not a required field when creating entities,  the community user would not have access to the case with out a community if it was reported by another user in the same community. Thus community users should always add the community to entities they need to have access to.
- They should not have the right to assign or edit epid number
- They should have full access to all entities with NA communities but are linked to facilities (or home addresses)  in the community of the user

**Comment by DavidBaldsiefen** [comment]
>     * we can call this role "community surveillance officer"

I would suggest just "Community officer", especially if he combines the rights of community surveillance & contact officer

>     * In order not to create another community contact officer role, we can  give this role the "community contact officer" rights.
> 
>     * they can have the combination of surveillance officer and contact officer rights but limited to the community of the user

Agreed

>     * Since the community is not a required field when creating entities,  the community user would not have access to the case with out a community if it was reported by another user in the same community. Thus community users should always add the community to entities they need to have access to.

Should we force them to fill that field? There might be situations where a user from community A needs to add a case/contact which does not belong to his communities responsibility (e.g. an event in my community, where participants from other communities took part. I should still be able to add all those participants to sormas, even though I won't be able to further process their cases or contacts)

>     * They should not have the right to assign or edit epid number

I don't know exactly what the epid number does but that can be implemented.

>     * They should have full access to all entities with NA communities but are linked to facilities (or home addresses)  in the community of the user

I will check out the implementation required for this. At the moment, they cannot see entities with NA communties.

I hope we can make this a quick discussion because it would be really nice to get this implemented for 1.57. If there are unclear points which are not critical, I would even think about putting them in another ticket and implementing this as-is with the changes mentioned above.

**Comment by DavidBaldsiefen** [comment]
@bernardsilenou You can see the rights I've now given to the Community Officer here:
https://github.com/hzi-braunschweig/SORMAS-Project/pull/4369/files

**Comment by bernardsilenou** [comment]
@DavidBaldsiefen  @MateStrysewske The rights looks OK.
I have 2 comments that that I think are not to be implemented in this ticket. If you think any of the points are OK to have, please comments so that I can create issues for them:
1. I only found DASHBOARD_CAMPAIGNS_ACCESS, what about default dashboard? I think we need to activate the default dashboard for them also.
2. It would also be good that we add the right to view archived task to surveillance officers and community officers. Only in the web app.

Comment: no automatic task goes to community officers for now, thus automatic case investigation right would not be used be CO. The right to investigate cases is still good fro CO to have because other users can assign investigation.

**Comment by bernardsilenou** [comment]
@DavidBaldsiefen  Comments on some of your questions:
`Should we force them to fill that field? There might be situations where a user from community A needs to add a case/contact which does not belong to his communities responsibility (e.g. an event in my community, where participants from other communities took part. I should still be able to add all those participants to sormas, even though I won't be able to further process their cases or contacts)`
If the reporting user is the CO, yes we force them as long as the responsible district of the entity is in their district, we can prefil the community as we have been doing. But if they create an entity in another district, then I do not think we should make community required. Community is not also requires for other users roles creating entities in the community of the CO, just facility is required. That is why i was suggesting that if other users create entities with NA community but those entities are linked to facilities with the same community as the CO, then the CO should  have full access to the entities (This is an additional rule that we can  handle in a future issue in case no time for it now)
![image](https://user-images.githubusercontent.com/12996724/107879302-53a14000-6ed8-11eb-8e74-34ff308df49f.png)

- For Epid number, there is a right for it and you did not activate it, so all is fine. 
- Every other need can be carried further, Let go on with implementation please, Thanks

**Comment by DavidBaldsiefen** [comment]
Thank you for the reply!

> I only found DASHBOARD_CAMPAIGNS_ACCESS, what about default dashboard? I think we need to activate the default dashboard for them also.

The default DASHBOARD_VIEW right will be removed with PR #4364 

> It would also be good that we add the right to view archived task to surveillance officers and community officers. Only in the web app.

Agreed for the community officer, but I would be careful with changing rights of existing roles

>  If the reporting user is the CO, yes we force them as long as the responsible district of the entity is in their district, we can prefil the community as we have been doing. But if they create an entity in another district, then I do not think we should make community required. Community is not also requires for other users roles creating entities in the community of the CO, just facility is required. That is why i was suggesting that if other users create entities with NA community but those entities are linked to facilities with the same community as the CO, then the CO should have full access to the entities (This is an additional rule that we can handle in a future issue in case no time for it now)

Alright, can you create an issue for that once this one is merged? Thank you!

**Comment by MateStrysewske** [comment]
@DavidBaldsiefen Seconding that we don't change anything about existing user roles in this issue.
